test-utils/shared/image.tsx

```diff
@@ -0,0 +1,88 @@
+import {render, screen} from '@testing-library/react';
+import React, {ElementType} from 'react';
+
+import {getQaAttrubutes} from '../../src';
+import {QAProps} from '../../src/models';
+import {isCompressible} from '../../src/utils/imageCompress';
+import {ERROR_INPUT_DATA_MESSAGE} from '../constants';
+
+export const testSourceProps = <T,>({
+    component: Component,
+    props,
+    options,
+}: {
+    component: ElementType;
+    props: T &
+        QAProps & {
+            src?: string;
+            tablet?: string;
+            desktop?: string;
+            mobile?: string;
+            disableCompress?: boolean;
+        };
+    options?: {
+        qaId?: string;
+    };
+}) => {
+    const src = props.src || props.desktop;
+    if (!src) {
+        throw new Error(ERROR_INPUT_DATA_MESSAGE);
+    }
+    const qaId = options?.qaId;
+
+    const qaAttributes = getQaAttrubutes(
+        props.qa,
+        'mobile-source-compressed',
+        'mobile-source',
+        'tablet-source-compressed',
+        'tablet-source',
+        'display-source-compressed',
```

Comment:
why display instead of desktop? and what about source without compress?

Reply:
agree